admin/admin.py

```diff
@@ -105,15 +125,51 @@ def save_settings(settings: Dict[str, Any], flash_msg: str) -> Response:
 
     :returns: Redirect with flash message
     """
-    config_file_path = path.join(app.config['APP_SHARED_FOLDER'], 'banner_settings.json')
+    config_file_path = path.join(app.config['APP_SHARED_FOLDER'], 'admin_settings.json')
     app.config.update(settings)
 
     try:
-        with open(config_file_path, 'w') as file:
-            json.dump(settings, file)
+        # Load existing settings and merge with new settings
```

Comment:
We could split reading and writing the settings. Then we can also more specific error handling.

```
# Load existing settings and merge with new settings
with open(config_file_path, 'r') as file:
    file_settings = json.load(file)
    file_settings.update(settings)

# Overwrite the file with the updated settings
with open(config_file_path, 'w') as file:
    json.dump(file_settings, file, indent=4)
```

Reply:
Suggestion adopted. Added new error handling